beluga/routes.py

```diff
@@ -85,46 +89,40 @@
     if lat or lon or radius:
         if not lat or not lon or not radius:
             raise abort(400, "search area must be fully specified")
-    
+
     # Acquire the events the user requested
     with session_scope() as db_session:
-        events = db_session.query(Event)
+        events = db_session.query(
+            Event.title,
+            Event.start_time,
+            Event.end_time,
+            Event.location.ST_AsGeoJSON().label("location_json")
+        )
+
         if start_time:
             events = events.filter(Event.start_time >= start_time)
             events = events.filter(Event.end_time <= end_time)
-        if lat:
-            # Convert from lat/lon/radius to upper/lower bounds for lat/lon
-
-            # We need to find the length of one degree longitude at the
-            # center point we were given. To do so, we take the cosine
-            # of the latitude we are given and multiply that by the length
-            # of one degree longitude at the equator.
-            lat_rads = (math.pi * float(lat)) / 180.0 # given lat in radians
-            lon_factor = 1.0 / ((math.pi / 180.0) * RADIUS_OF_EARTH * math.cos(lat_rads))
-
-            # Using an approximation here for simplicity.
-            # The distance of a single degree of latitude varies around
-            # the globe, but does not vary as dramatically as longitude.
-            # We therefore can use an approximation here. More info at
-            # https://en.wikipedia.org/wiki/Latitude#Length_of_a_degree_of_latitude.
-            lat_factor = 1.0 / 111.0 # 1 deg lat ~= 111 km (give or take 1 km)
 
-            min_lon = float(lon) - (float(radius) * lon_factor)
-            max_lon = float(lon) + (float(radius) * lon_factor)
-
-            min_lat = float(lat) - (float(radius) * lat_factor)
-            max_lat = float(lat) + (float(radius) * lat_factor)
-
-            # TODO once we can query by location, the query
-            # should be modified here to reflect the location we 
-            # got from the client.
-            pass
-
-    # Sort by start_time for now.
-    events = events.order_by(Event.start_time.asc())
+        if lat:
+            events = events.filter(
+                Event.location.ST_DWithin(
+                    WKTElement("POINT({} {})".format(lon, lat)), # Center point
+                    (float(radius) * METERS_PER_KILOMETER) # Convert from km to m
+                )
+            )
+
+        events = events.order_by(Event.start_time.asc())
```

Comment:
Why don't we replace this with something like:

```python
events = events.order_by(
    func.ST_Distance(
        Event.location,
        WKTElement('Point({} {})'.format(lat, lon)))
)
```

And we're ordered by distance from our centroid :tada:

Reply:
See above ☝️